### PR TITLE
Cover/cmds/core/sync

### PR DESCRIPTION
**Pull request by llogen**
Refactored and added tests.

Signed-off-by: llogen <christophlange95@googlemail.com>

### CODE REVIEW

File: cmds/core/sync/sync.go [hunk 1]
```diff
@@ -28,39 +28,44 @@ var (
...
 }
 
-func main() {
-	flag.Parse()
+func sync(args []string) error {
 	switch {
 	case *data:
-		doSyscall(unix.SYS_FDATASYNC)
+		err := doSyscall(unix.SYS_FDATASYNC, args)
+		return err
```
Comment (MDr164) [inline cmds/core/sync/sync.go:56]: You can simplify this: `return doSyscall(unix.SYS_FDATASYNC, args)`

File: cmds/core/sync/sync.go [hunk 2]
```diff
@@ -28,39 +28,44 @@ var (
...
 	case *filesystem:
-		doSyscall(unix.SYS_SYNCFS)
+		err := doSyscall(unix.SYS_SYNCFS, args)
+		return err
```
Comment (MDr164) [inline cmds/core/sync/sync.go:59]: same as above

File: cmds/core/sync/sync.go [hunk 3]
```diff
@@ -28,39 +28,44 @@ var (
...
 		f, err := os.OpenFile(fileName, syscall.O_RDONLY|syscall.O_NOCTTY|syscall.O_CLOEXEC, 0o644)
 		if err != nil {
-			log.Fatal(err)
+			return err
 		}
 		_, _, err = syscall.Syscall(syscallNum, uintptr(f.Fd()), 0, 0)
-		if err != nil {
-			log.Fatal(err)
+		if err.Error() != "errno 0" {
+			return err
```
Comment (ChriMarMe) [inline cmds/core/sync/sync.go:45]: Das sieht mir nicht ganz sauber aus. Koennte man `!errors.Is(...)` verwenden?

File: cmds/core/sync/sync_test.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+			name:   "default",
+			input:  []string{file1.Name(), file2.Name()},
+			expErr: nil,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			*data = tt.data
+			*filesystem = tt.filesystem
+			err := sync(tt.input)
+			if err != nil {
```
Comment (MDr164) [inline cmds/core/sync/sync_test.go:65]: Should be `if err := sync(tt.input); err != nil {`

**Review by MDr164 — CHANGES_REQUESTED** [review]

File: cmds/core/sync/sync.go (hunk 3 above)
Comment (llogen) [reply cmds/core/sync/sync.go:45]: Oops, missed that syscall error can be checked against fs errors.
Thanks for that catch!

**Review by walterchris — APPROVED** [review]

File: cmds/core/sync/sync_test.go [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+			input:  []string{file1.Name(), file2.Name()},
+			expErr: nil,
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			*data = tt.data
+			*filesystem = tt.filesystem
+			if err := sync(tt.input); err != nil {
+				if tt.expErr.Error() != err.Error() {
+					t.Errorf("Expected error: %v, got: %v", tt.expErr, err)
```
Comment (walterchris) [inline cmds/core/sync/sync_test.go:66]: Can you change this according to: https://github.com/u-root/u-root/blob/main/CONTRIBUTING.md#unit-test-checks

**Review by walterchris — CHANGES_REQUESTED** [review]

**Review by walterchris — APPROVED** [review]
LGTM

**Pull request closed without merging**